### PR TITLE
feat: introduce small-file picker for hummock

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://raw.githubusercontent.com/risingwavelabs/risingwave/17af8a747593ebdbfa826691daf75bdab7d14fa0/.github/contributor-license-agreement.txt).

## Summary

Add `SmallFile` compaction type to merge small SST files in LSM tree's last level.

## Implementation

- **Picker**: `SmallFileCompactionPicker` with 3 strategies:
  - Greedy selection: collect all consecutive small files
  - Minimum count: require at least 3 files
  - Vnode boundary: don't cross vnode boundaries
- **Selector**: `SmallFileCompactionSelector` wrapper
- **Config**: Add `periodic_small_file_compaction_interval_sec` (default: 300s)
- **Tests**: 11 comprehensive unit tests covering all edge cases

## Rationale

- Only scans last level (90%+ data resides there)
- Reduces file count and improves scan performance
- Respects vnode boundaries for partition isolation

## Checklist

- [ ] I have written necessary rustdoc comments.
- [ ]  I have added necessary unit tests and integration tests.
- [ ]  I have added test labels as necessary. 
- [ ]  I have added fuzzing tests or opened an issue to track them. 
- [ ]  My PR contains breaking changes. 
- [ ]  My PR changes performance-critical code, so I will run (micro) benchmarks and present the results. 
- [ ]  I have checked the [Release Timeline](https://github.com/risingwavelabs/rw-commits-history/blob/main/release_timeline.md) and [Currently Supported Versions](https://docs.risingwave.com/changelog/release-support-policy#support-end-dates-for-recent-releases) to determine which release branches I need to cherry-pick this PR into. 

## Documentation

- [ ]  My PR needs documentation updates. 

<details>
<summary><b>Release note</b></summary>

</details>

### CODE REVIEW

File: src/meta/src/hummock/compaction/picker/small_file_compaction_picker.rs [hunk 1]
```diff
@@ -0,0 +1,1139 @@
...
+            state.clear();
+            return None;
+        }
+
+        // Successfully selected enough files, update end_bound for next round
+        let select_last_sst = select_input_ssts.last().unwrap();
+        state.skip_to_sst_right(select_last_sst);
+
+        Some(CompactionInput {
+            select_input_size: select_input_ssts.iter().map(|sst| sst.sst_size).sum(),
```
Comment (zwang28) [inline src/meta/src/hummock/compaction/picker/small_file_compaction_picker.rs:233]: Could we assert that the select_input_ssts are always consecutive?

**Review by zwang28 — APPROVED** [review]

**Review by zwang28 — CHANGES_REQUESTED** [review]
Discussed offline. There's a known issue selecting multiple files.
